### DISCUSSION TITLE
Heltec V3 boards

### DISCUSSION THREAD

**Discussion by turbocadillac** [spleenware/ripple#65] [General]
Since the v2 board is discontinued and hard to find is it possible to use the v3 board?

**Comment by spleenware** [comment]
Yes, I should be able to produce a .bin for the V3 board. I don't have the various parts needed at hand right now, so you can be  the guinea pig :-)  I'll upload the .bin a bit later, and post another reply here when it's ready.

**Comment by turbocadillac** [comment]
Thank you so much! I'll gladly be the guinea pig! I'll try and document as much as possible and report back to you! I sent you a some btc as well so you can get a coffee!

**Comment by spleenware** [comment]
OK, I've uploaded an experimental .bin for the V3:
   https://github.com/spleenware/ripple/blob/master/RippleQwertyGeo-BB-HeltecV3.bin

For the wiring:
  for the wake button -> pin 4 (and other to 3.3V)
  buzzer + -> pin 2
  buzzer - -> GND
  GPS tx -> pin 7
  GPS rx -> pin 6
  GPS vcc -> Vext
  GPS Gnd -> GND
  BB keyboard, SCL -> pin 18 (on back of board)
  BB keyboard, SDA -> pin 17 (on back of board)
  BB keyboard, vcc -> 3.3V
  BB keyboard, Gnd -> GND

**Comment by turbocadillac** [comment]
Thank you! so i flashed it with the newest version of esptool.py i have one completely soldered up and one board that isn't. they just go black and nothing shows up, the keyboard doesn't light up or the power led on the gps. same with the unsoldered board.

**Reply by spleenware** [reply]
Did you flash the firmware .bin to offset 0x10000 ? If not, you might have to overwritten the bootloader (at offset 0x0).
There is a web-based tool here: https://adafruit.github.io/Adafruit_WebSerial_ESPTool/
Once connected to the board, you can enter 10000 as the offset, then select the firmware .bin file, then click 'Program'.

**Comment by turbocadillac** [comment]
Here is what i used the first time, esptool.py --port /dev/tty.SLAB_USBtoUART --chip ESP32-S3 write_flash -z 0x1000 /Users/andrewkoehler/Downloads/RippleQwertyGeo-BB-HeltecV3.bin. That web-based tool is great!  They are both still black though should i attempt to flash the bootloader?

**Reply by spleenware** [reply]
Yes, I think you have overwritten the bootloader and the partition table :-(
I have uploaded a zip that should have the standard binaries (eg. bootloader.bin, partitions.bin, etc), and have prefixed each file with the hex offset each should be flashed to:
   https://github.com/spleenware/ripple/blob/master/HeltecV3-restore.zip

The app binary should be flashed to offset 0x10000 (four zeroes)

**Comment by turbocadillac** [comment]
Thank you for that! I was able to salvage one board! So far everything seems to be working , but when it goes to sleep it won't wake back up with the button. It also won't communicate with the older v2 board with the rtc, when I exchange keys the v3 board will add the users but the v2 won't.

**Reply by spleenware** [reply]
Have you wired up the GPS yet? It needs that to initialise the internal clock. That could be one issue.
So, both V3 and V2 can do the Exchange stuff? Make sure both have the same frequency and spreading factor (the default was 10, but is now 9, I think)

**Reply by spleenware** [reply]
Regarding the wake button, have you wired like this?
![HeltecV3-wake-btn](https://github.com/spleenware/ripple/assets/1137445/c887652f-f403-4dd0-b663-030dd89822e8)

**Comment by turbocadillac** [comment]
Yes but I also added a 10k resistor. I will open it up and cut it out. Ok I will look into that. The v2 boards don't have gps just the real time clocks, I'll order more gps modules

**Reply by turbocadillac** [reply]
I removed the resistor and it works perfect

**Reply by turbocadillac** [reply]
so i removed the real time clock out of the v2 and installed my extra gps module and flashed RippleQwertyGeo-HeltecV2-BB, and i can get the v3 to exchange key with v2 and the contact comes up on the screen of the v3 but not on the v2. I also changed the SF to 9 on the v3 as well with no luck
